back/sssscs/src/main/java/com/ib/certificate/CertificateController.java

```diff
@@ -0,0 +1,128 @@
+import com.ib.certificate.dto.CertificateSummaryItemDto;
+import com.ib.user.IUserService;
+import com.ib.user.User;
+
+@RestController
+@RequestMapping("/api/cert")
+public class CertificateController {
+	@Autowired
+	private ICertificateService certificateService;
+	
+	@Autowired
+	private ICertificateRequestService certificateRequestService;
+	
+	@Autowired
+	private IUserService userService;
+	
+	@PostMapping("/request")
+	public ResponseEntity<String> makeRequest(/*@DTO(CertificateRequestCreateDto.class)*/ @RequestBody CertificateRequestCreateDto certificate) {
+		// TODO: Automatic mapping.
+		
+		CertificateRequest req = new CertificateRequest();
+		req.setIssuer(userService.findById(certificate.getIssuerId()));
+		req.setType(certificate.getType());
+		req.setValidTo(certificate.getValidTo());
+		req.setStatus(Status.PENDING);
+		if (certificate.getParentId() != 0) {
+			Certificate parent = certificateService.findById(certificate.getParentId());
+			
+			// TODO: 404 if parent is null.
+			
+			req.setParent(parent);
+		}
+		
+		req = certificateRequestService.makeRequest(req);
+		
+		if (certificateRequestService.canAutoAccept(req)) {
+			certificateService.accept(req);
+		}
+		
+		return ResponseEntity.ok(req.toString());
+	}
+	
+	@PutMapping("/request/accept/{id}")
+	public ResponseEntity<String> acceptRequest(@PathVariable Long id) {
+		CertificateRequest req = certificateRequestService.findByIdAndStatusEquals(id, Status.PENDING);
+		if (req == null) {
+			return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+		}
+		
+		Certificate cert = certificateService.accept(req);
+		return ResponseEntity.ok(cert.toString());
+	}
+	
+	@GetMapping("/request/{id}")
+	public ResponseEntity<List<CertificateRequestDto>> getMyRequests(@PathVariable Long id) {
+		User issuer = userService.findById(id);
+		if (issuer == null) {
+			return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+		}
+		
+		List<CertificateRequest> requests = certificateRequestService.findByIssuer(issuer);
+		List<CertificateRequestDto> result = requests.stream()
+				.map(r -> new CertificateRequestDto(
+						r.getId(), 
+						r.getIssuer().getId(), 
+						r.getValidTo(), 
+						r.getParent() != null ? r.getParent().getId() : 0, 
+						r.getType(), 
+						r.getStatus())
+		).toList();
+		
+		return ResponseEntity.ok(result);
+	}
+	
+	@GetMapping("/request/incoming/{id}")
+	public ResponseEntity<List<CertificateRequestDto>> getPendingRequestsIssuedTo(@PathVariable Long id) {
+		User issuee = userService.findById(id);
+		if (issuee == null) {
+			return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+		}
+		
+		List<CertificateRequest> requests = certificateRequestService.findByIssuee(issuee);
+		List<CertificateRequestDto> result = requests.stream()
+				.map(r -> new CertificateRequestDto(
+						r.getId(), 
+						r.getIssuer().getId(), 
+						r.getValidTo(), 
+						r.getParent() != null ? r.getParent().getId() : 0, 
+						r.getType(), 
+						r.getStatus())
```

Comment:
Consider using `ModelMapper` for this

Reply:
Done in [d4eb49d](https://github.com/magley/sssscs/pull/3/commits/d4eb49df6f8d640946eccd94b08b62f12e6842b1)